### PR TITLE
use `node.error` for CSS relevant errors

### PR DESCRIPTION
**Pull request by romainmenke**
fixes https://github.com/postcss/postcss-import/issues/518

I've kept the `new Error()` when checking plugin options.
I don't think these are CSS syntax errors : https://github.com/postcss/postcss/blob/main/docs/guidelines/plugin.md#41-use-nodeerror-on-css-relevant-errors

### CODE REVIEW

File: lib/resolve-id.js [hunk 1]
```diff
@@ -32,10 +32,10 @@ module.exports = function resolveId(id, base, options) {
     .catch(() => {
       if (paths.indexOf(base) === -1) paths.unshift(base)
 
-      throw new Error(
+      throw node.error(
         `Failed to find '${id}'
   in [
-    ${paths.join(",\n        ")}
+    ${paths.join(",\n    ")}
```
Comment (romainmenke) [inline lib/resolve-id.js:38]: Indent before :

```
    message: `postcss-import: <css input>:3:1: Failed to find 'layer\`\`\`.css'␊
      in [␊
        /projects/postcss-import,␊
            /projects/postcss-import/test/fixtures/imports␊
      ]`,
```

Indent now :

```
    message: `postcss-import: <css input>:3:1: Failed to find 'layer\`\`\`.css'␊
      in [␊
        /projects/postcss-import,␊
        /projects/postcss-import/test/fixtures/imports␊
      ]`,
```

File: test/layer.js [hunk 2]
```diff
@@ -49,7 +49,9 @@ test("should error when value is not a function", t => {
   return postcss()
     .use(atImport({ nameLayer: "not a function" }))
     .process("", { from: undefined })
-    .catch(error => t.is(error.message, "nameLayer option must be a function"))
+    .catch(error =>
+      t.regex(error.message, /nameLayer option must be a function/s)
```
Comment (RyanZim) [inline test/layer.js:53]:
```suggestion
      t.regex(error.message, /nameLayer option must be a function/)
```

File: test/plugins.js [hunk 3]
```diff
@@ -35,7 +35,7 @@ test("should error when value is not an array", t => {
   return postcss()
     .use(atImport({ plugins: "foo" }))
     .process("", { from: undefined })
-    .catch(error => t.is(error.message, "plugins option must be an array"))
+    .catch(error => t.regex(error.message, /plugins option must be an array/s))
```
Comment (RyanZim) [inline test/plugins.js:38]:
```suggestion
    .catch(error => t.regex(error.message, /plugins option must be an array/))
```

**Review by RyanZim — COMMENTED** [review]
Don't need `/s` flag in new test regexes

**Review by RyanZim — APPROVED** [review]

**Pull request merged**